### PR TITLE
Add the List, Stack and Node implementations and tests, and add the r…

### PR DESCRIPTION
**Pull request by Larwive**
…equirement tracker.

### CODE REVIEW

**Comment by ttrenty** [comment]
Great!

How to start the tests please?

Could you replace the current `echo "python -m pytest"` line in the `.github/workflows/CI-PR-dev.yml` file with the command(s) to start your tests please so that the tests are run in the pipeline too?

**Comment by Larwive** [comment]
To start the tests, do as following :

1. Check is `pytest` is installed. If it's not, upgrade `pip` with `pip3 install --upgrade pip` then `pip3 install pytest`.
2. Check if the `src` directory containing the `visualgo` package is in the `PYTHONPATH` environment variable.
3. Run the tests with `pytest tests/visualgo/types` from the root of the repository.

I'll try to update `.github/workflows/CI-PR-dev.yml`.

**Comment by Larwive** [comment]
Just created a pull request for it.
I used `python -m unittest discover tests/visualgo/types` instead of using `pytest`. Probably because `pytest` needs to be installed separately from Python.

Both work well for us.

**Review by Cax36940 — DISMISSED** [review]

File: tests/visualgo/types/test_Node.py
Comment (ttrenty) [inline tests/visualgo/types/test_Node.py:1]: Comments should be changed in English

File: tests/visualgo/types/test_List.py
Comment (ttrenty) [inline tests/visualgo/types/test_List.py:1]: Missing some tests for other methods

Comment (ttrenty) [reply tests/visualgo/types/test_List.py:1]: Can be done on another PR

**Review by ttrenty — APPROVED** [review]
More unit tests can be added for List in a latter PR

**Review by ttrenty — APPROVED** [review]

**Review by Cax36940 — APPROVED** [review]

**Pull request merged**